src/AzureClient/AzureExecutionTarget.cs

```diff
@@ -3,132 +3,167 @@
 
 #nullable enable
 
-using System;
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.Azure.Quantum;
 using Microsoft.Quantum.QsCompiler;
 using Microsoft.Quantum.Runtime.Submitters;
 
-namespace Microsoft.Quantum.IQSharp.AzureClient
+namespace Microsoft.Quantum.IQSharp.AzureClient;
+
+internal enum AzureProvider
 {
-    internal enum AzureProvider
+    IonQ,
+    Quantinuum,
+    // NB: This provider name is deprecated, but may exist in older
+    //     workspaces and should still be supported.
+    Honeywell,
+    QCI,
+    Microsoft,
```

Comment:
(nit) Consider placing MS first. MS seems to me mandatory, whereas all others seem optional.

Reply:
Easy enough, let me go change that, then.